src/utils/sst.ts

```diff
@@ -0,0 +1,188 @@
+    );
+  } else {
+    await writeFile(
+      resolve(nitro.options.rootDir, "sst.config.ts"),
+      sstConfigTemplate({
+        edge: nitro.options.awsLambda?.target === "edge",
+        sstOptions: nitro.options.awsLambda?.sstOptions,
+        sstDir,
+      })
+    );
+  }
+
+  nitro.logger.info(
+    `
+    Generated SST app !
+    Install SST dependencies: "npm i -D sst aws-cdk-lib constructs"
+
+        - Dev: "npx sst dev"
+        - Build: "npx sst build"
+        - Deploy: "npx sst deploy"
+        - Remove: "npx sst remove"
+        - Console: "npx sst console
+    `
+  );
+}
+
+const sstConfigTemplate = ({
+  edge = false,
+  sstOptions,
+  sstDir,
+}: {
+  edge: boolean;
+  sstOptions: PresetOptions["awsLambda"]["sstOptions"];
+  sstDir: string;
+}) =>
+  /* ts */ `
+import { SSTConfig } from "sst";
+import { NitroApp } from "${sstDir}";
+
+export default {
+  config(_input) {
+    return {
+      ${optionsToString({ name: "nitro", region: "us-east-1", ...sstOptions })}
+    };
+  },
+  stacks(app) {
+    app.stack(function Site({ stack }) {
+      const app = new NitroApp(stack, "app", {
+        path: "./",
+        edge: ${edge ? "true" : "false"},
+      });
+
+      stack.addOutputs({
+        url: app.url,
+      });
+    });
+  }
+} satisfies SSTConfig;`.trim();
+
+function optionsToString(
+  obj?: PresetOptions["awsLambda"]["sstOptions"],
+  indent = "      "
+): string {
+  if (!obj) {
+    return "";
+  }
+  return Object.entries(obj)
+    .map(([key, value]) => {
+      return typeof value === "object" && value !== null
+        ? `${key}: {\n${optionsToString(value, indent + "  ")}\n${indent}}`
+        : `${key}: ${JSON.stringify(value)}`;
+    })
+    .join(",\n" + indent);
+}
+
+const nitroAppTemplate = () =>
+  /* ts */ `
+import fs from "node:fs"
+import path from "node:path";
+import type { Function as CdkFunction } from "aws-cdk-lib/aws-lambda";
+
+import { SsrSite } from "sst/constructs/SsrSite.js";
+import { SsrFunction } from "sst/constructs/SsrFunction.js";
+import { EdgeFunction } from "sst/constructs/EdgeFunction.js";
+
+export class NitroApp extends SsrSite {
+  public getConstructMetadata() {
+    return {
+      type: "NitroApp" as const,
+      ...this.getConstructMetadataBase(),
+    };
+  }
+
+  protected initBuildConfig() {
+    return {
+      typesPath: "./",
+      serverBuildOutputFile: ".output/server/index.mjs",
+      clientBuildOutputDir: ".output/public",
+      clientBuildVersionedSubDir: ".output",
+    };
+  }
+
+  protected validateBuildOutput() {
+    const serverDir = path.join(this.props.path, ".output/server");
+    const clientDir = path.join(this.props.path, ".output/public");
+    if (!fs.existsSync(serverDir) || !fs.existsSync(clientDir)) {
+      throw new Error(
+        'Build output inside ".output/" does not contain the "server" and "public" folders.'
+      );
+    }
+
+    super.validateBuildOutput();
+  }
+
+  protected createFunctionForRegional(): CdkFunction {
```

Comment:
This template may need to be updated for the [changes that were made to SST ](https://github.com/sst/sst/blob/c61396b0a9197dcac59bc3ffd5bdbdc308b47f30/packages/sst/src/constructs/SsrSite.ts).

It has been updated in my [test repo.](https://github.com/jdevdevdev/nuxt-sst/blob/d4da6aad6313dd6e4e7a20dd59509fd4bfe6dd0a/stacks/NuxtSite.ts)

A faster way to merging might be to split SST integration and the lambda edge preset code into different PRs. Let the user build the deployment and add deployment guides / documentation on nitro.unjs.io / nuxt.com